http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -516,6 +523,150 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
       }
     } yield ()
 
+  private def monitorDeleteDisk(params: PollDeleteDiskParams)(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      wsmApi <- buildWsmControlledResourceApiClient
+      getWsmJobResult = F.delay(wsmApi.getDeleteAzureDiskResult(params.workspaceId.value, params.jobId.value))
+
+      _ <- F.sleep(
+        config.deleteDiskPollConfig.initialDelay
+      )
+
+      resp <- streamFUntilDone(
+        getWsmJobResult,
+        config.deleteDiskPollConfig.maxAttempts,
+        config.deleteDiskPollConfig.interval
+      ).compile.lastOrError
+
+      _ <- resp.getJobReport.getStatus match {
+        case JobReport.StatusEnum.FAILED =>
+          for {
+            _ <- logger.error(s"Wsm deleteDisk job failed due to ${resp.getErrorReport.getMessage}")
+            _ <- params.diskId.traverse(id =>
+              dbRef.inTransaction(persistentDiskQuery.updateStatus(id, DiskStatus.Error, ctx.now))
+            )
+            _ <- F.raiseError[Unit](
+              AzureRuntimeDeletionError(
+                params.runtime.id,
+                params.workspaceId,
+                s"Wsm deleteDisk job failed due to ${resp.getErrorReport.getMessage}, disk resource id ${params.wsmResourceId}"
+              )
+            )
+          } yield ()
+        case JobReport.StatusEnum.RUNNING =>
+          params.diskId.traverse(id =>
+            dbRef.inTransaction(persistentDiskQuery.updateStatus(id, DiskStatus.Error, ctx.now))
+          ) >> F.raiseError[Unit](
+            AzureRuntimeDeletionError(
+              params.runtime.id,
+              params.workspaceId,
+              s"Wsm deleteDisk was not completed within ${config.deleteDiskPollConfig.maxAttempts} attempts with ${config.deleteDiskPollConfig.interval} delay, disk resource id ${params.wsmResourceId}"
+            )
+          )
+        case JobReport.StatusEnum.SUCCEEDED =>
+          for {
+            _ <- logger.info(ctx.loggingCtx)(
+              s"disk  for runtime ${params.runtime.id} with resource id ${params.wsmResourceId} is deleted successfully"
+            )
+          } yield ()
+      }
+    } yield ()
+
+  private def monitorDeleteVm(params: PollVmParams)(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      wsmApi <- buildWsmControlledResourceApiClient
+      getWsmJobResult = F.delay(wsmApi.getDeleteAzureVmResult(params.workspaceId.value, params.jobId.value))
+
+      _ <- F.sleep(
+        config.deleteVmPollConfig.initialDelay
+      )
+
+      resp <- streamFUntilDone(
+        getWsmJobResult,
+        config.deleteVmPollConfig.maxAttempts,
+        config.deleteVmPollConfig.interval
+      ).compile.lastOrError
+
+      _ <- resp.getJobReport.getStatus match {
```

Comment:
This is the smallest nit ever, but could you keep the order of the statuses the same throughout? e.g. success, then failure then running?